Add AMD ROCm support

mhorst00 opened the pull request:
This pull request adds support for AMDs ROCm GPU libraries. It targets ROCm versions >=5.0. Integrated documentation for doxygen has been added, though documentation in the wiki will be missing for now. 

I have merged the latest state of the LIKWID repo into this pretty old fork, so there may be some slight issues, especially around NVIDIA code changed after May of 2021.

Let me know if there are any changes needed.

TomTheBear commented:
Thanks for the PR. How do we want to proceed?

I enabled the Github Actions for the PR and as you can see, it does not compile. The error is unrelated to Nvidia GPUs.

mhorst00 commented:
My imports in luawid.c got messed up by automated sorting, I fixed it and it now compiles for me with either NVIDIA, AMD or no GPU support enabled.

mhorst00 commented:
I had some import conflicts because my formatter sorted them wrong. I now fixed this and rebased on current master again. I can compile this branch with AMD or NVIDIA support or without GPU support.
